### PR TITLE
Story5 - Filtered cars by make 

### PR DESCRIPTION
**Pull request by Alina1319**
Display a single filter button for each Make
Clicking a Make filter button displays only the cars in that Make
Filter buttons must be displayed in the order of most cars to least cars

### CODE REVIEW

File: css/styles.css [hunk 1]
```diff
@@ -158,4 +160,117 @@ main {
...
     border-radius: 100px;
+}
+
+.make-btn {
+    background-color: #ffffff;
+    text-decoration: none;
+    border-radius: 4px;
+    padding: 5px 8px;
+    margin: 5px;
+    border: 1px solid #000000FF;
```
Comment (mporam) [inline css/styles.css]: is this supposed to be an alpha hex? if so, can we use RGBA please

File: src/ViewHelpers/CarViewHelper.php [hunk 2]
```diff
@@ -14,11 +14,11 @@ private static function displayCar(CarEntity $car): string
             '<div class="brand-model-container"><h1 class="makeCard">' . $car->getMake() . '</h1>' .
             '<h1 class="modelCard">' . $car->getModel() . '</h1></div>' .
             '<div class="link"><a class="index-link" href="details.php?id='.$car->getId().'">More Details</a></div></div>';
-
         return $output;
     }
 
     public static function displayCars(array $cars): string
+
```
Comment (mporam) [inline src/ViewHelpers/CarViewHelper.php]: why?

File: src/ViewHelpers/FiltersViewHelper.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace KoalaCars\ViewHelpers;
+
+class FiltersViewHelper
+{
+    public static function displayMakes(array $makes, string $selectedMake): string
+    {
+        $output = '<div class="make-container">';
+        foreach ($makes as $make) {
+            foreach ($make as $value) {
```
Comment (mporam) [inline src/ViewHelpers/FiltersViewHelper.php]: why? u dont need this

File: src/ViewHelpers/FiltersViewHelper.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                if ($value) {
+                    $class = '';
+                    if ($selectedMake == $value) {
+                        $class = ' active';
+                    }
+                    $output .= '<button class="make-btn ' . $class . '"><a href="index.php?make=' . $value . '">' . $value . '</a></button>';
+                } else {
+                    return 'invalid information';
```
Comment (mporam) [inline src/ViewHelpers/FiltersViewHelper.php]: do you really want the user to see this message if something in the db has gone wrong?

File: src/ViewHelpers/FiltersViewHelper.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                }
+            }
+        }
+        $output .= '</div> ';
+        $class = '';
+        if (empty($selectedMake)) {
+            $class = ' active';
+        }
+        $output .= '<button class="all-cars-btn ' . $class . '"><a href="index.php">All cars</a></button>';
```
Comment (mporam) [inline src/ViewHelpers/FiltersViewHelper.php]: this feels like it should be its own method

**Review by mporam — CHANGES_REQUESTED** [review]

File: css/styles.css (hunk 1 above)
Comment (SniffleSneeze) [reply css/styles.css]: Mistake , there is no alfa involve this will be resolve :)

File: src/ViewHelpers/FiltersViewHelper.php [hunk 6]
```diff
@@ -8,20 +8,22 @@ public static function displayMakes(array $makes, string $selectedMake): string
...
-            foreach ($make as $value) {
-                if ($value) {
-                    $class = '';
-                    if ($selectedMake == $value) {
-                        $class = ' active';
-                    }
-                    $output .= '<button class="make-btn ' . $class . '"><a href="index.php?make=' . $value . '">' . $value . '</a></button>';
-                } else {
-                    return 'invalid information';
+            if ($make) {
```
Comment (mporam) [inline src/ViewHelpers/FiltersViewHelper.php]: why? what do you think this if statement does??

**Review by mporam — CHANGES_REQUESTED** [review]

File: src/ViewHelpers/FiltersViewHelper.php [hunk 7]
```diff
@@ -8,13 +8,11 @@ public static function displayMakes(array $makes, string $selectedMake): string
     {
         $output = '<div class="make-container">';
         foreach ($makes as $make) {
-            if ($make) {
                 $class = '';
```
Comment (mporam) [inline src/ViewHelpers/FiltersViewHelper.php]: now your indentation is wrong

File: src/ViewHelpers/FiltersViewHelper.php [hunk 8]
```diff
@@ -8,13 +8,11 @@ public static function displayMakes(array $makes, string $selectedMake): string
     {
```
Comment (mporam) [inline src/ViewHelpers/FiltersViewHelper.php:8]: the unit tests for this method also fail

**Review by mporam — CHANGES_REQUESTED** [review]

**Review by mporam — APPROVED** [review]

**Pull request merged**